filters/filter.go

```diff
@@ -56,11 +56,32 @@ type Filter interface {
 }
 
 // NewFilter supports startsWith, endsWith, contains and a single wildcard
-// TODO(martinm): add rest of glob matching support and negation
+// along with negation
+// TODO(martinm): add rest of glob matching support
 func NewFilter(pattern string) (Filter, error) {
-	idx := strings.Index(pattern, WildcardChar)
+	nIdx := strings.Index(pattern, negationChar)
+	if nIdx != 0 {
```

Comment:
I think we should return an error if `nIdx > 0` because we don't allow "!" in tag name/values or for legacy statsd metrics. Thoughts?